db/migrate/20170725111950_change_enterprise_contact.rb

```diff
@@ -0,0 +1,18 @@
+class ChangeEnterpriseContact < ActiveRecord::Migration
+  def up
+    remove_columns :enterprises, :email, :contact
+
+    add_column :enterprises, :contact_id, :integer
+    add_foreign_key :enterprises, :spree_users, column: :contact_id
+
+    Enterprise.update_all 'contact_id = owner_id'
```

Comment:
This needs to respect the existing settings in the `email` field. As per our other discussion, the `email` of each enterprise should correspond to a `Spree::User` already. So we need to use `email` value to find the `contact_id` here.